ddtrace/internal/telemetry/writer.py

```diff
@@ -59,65 +101,30 @@ def __init__(self, agent_url=None):
         self._debug = asbool(os.environ.get("DD_TELEMETRY_DEBUG", "false"))
         forksafe.register(self._fork_writer)
 
-        self._headers = {
-            "Content-type": "application/json",
-            "DD-Telemetry-API-Version": "v1",
-            "DD-Telemetry-Debug-Enabled": str(self._debug).lower(),
-        }  # type: Dict[str, str]
-        additional_header_str = os.environ.get("_DD_TELEMETRY_WRITER_ADDITIONAL_HEADERS")
-        if additional_header_str is not None:
-            self._headers.update(parse_tags_str(additional_header_str))
-
-        # _sequence is a counter representing the number of requests sent by the writer
-        self._sequence = 1  # type: int
-
-    @property
-    def url(self):
-        return "%s/%s" % (self._agent_url, self.ENDPOINT)
-
-    def _send_request(self, request):
-        # type: (Dict) -> httplib.HTTPResponse
-        """Sends a telemetry request to the trace agent"""
-        with StopWatch() as sw:
-            try:
-                conn = get_connection(self._agent_url)
-                rb_json = self._encoder.encode(request)
-                conn.request("POST", self.ENDPOINT, rb_json, self._create_headers(request["request_type"]))
-
-                resp = get_connection_response(conn)
-                log.debug(
-                    "sent %d in %.5fs to %s/%s. response: %s",
-                    len(rb_json),
-                    sw.elapsed(),
-                    self._agent_url,
-                    self.ENDPOINT,
-                    resp.status,
-                )
-                return resp
-            finally:
-                conn.close()
+        # counter representing the number of events sent by the writer
+        self._sequence = itertools.count(1)
```

Comment:
Just a reminder that atomicity here is an implementation detail of `itertools.count` and that this could change at any time.